### PR TITLE
removed user read call in sessions and added organization key to model

### PR DESCRIPTION
**Pull request by rakeshSgr**

### CODE REVIEW

File: src/services/users.js [hunk 1]
```diff
@@ -127,6 +165,8 @@ module.exports = class UserHelper {
 			competency: userDetails.competency,
 			designation: userDetails.designation,
 			language: userDetails.language,
+			organization_name: userDetails.organization.name,
+			image: userDetails.image,
```
Comment (aks30) [inline src/services/users.js:169]: @rakeshSgr we can add this in column i am thinking, anyways in ELEVATE we have user image

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 11 Oct

File: src/database/migrations/20240813060340-update-sessions-columns-drop-views-create-again copy.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+'use strict'
+require('module-alias/register')
+require('dotenv').config()
+const materializedViewsService = require('@generics/materializedViews')
+
+/** @type {import('sequelize-cli').Migration} */
```
[the remark this answers is not in the document]
Comment (rakeshSgr) [reply src/database/migrations/20240813060340-update-sessions-columns-drop-views-create-again copy.js:6]: @aks30 its invalid file, not required

File: src/services/users.js (hunk 1 above)
Comment (rakeshSgr) [reply src/services/users.js:169]: ok

File: src/constants/interface-routes/sunbird-mentoring.json [hunk 3]
```diff
@@ -0,0 +1,3335 @@
+{
```
Comment (aks30) [inline src/constants/interface-routes/sunbird-mentoring.json:1]: @rakeshSgr Let's move these files in project root, in sunbird-mentoring and mentoring

File: src/database/migrations/20241007083318-add organisation column to OrgExt.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+require('module-alias/register')
+require('dotenv').config({ path: '../../.env' })
+const userRequests = require('@requests/user')
+
+module.exports = {
+	up: async (queryInterface, Sequelize) => {
+		try {
+			const { STRING } = Sequelize
+
+			await queryInterface.addColumn('organization_extension', 'organization_name', {
```
Comment (aks30) [inline src/database/migrations/20241007083318-add organisation column to OrgExt.js:11]: @rakeshSgr let's call it name only

File: src/database/migrations/20241007083318-add organisation column to OrgExt.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
+'use strict'
+require('module-alias/register')
+require('dotenv').config({ path: '../../.env' })
```
Comment (aks30) [inline src/database/migrations/20241007083318-add organisation column to OrgExt.js:3]: @rakeshSgr Rename file to just add name column to org extension

File: src/database/models/organizationExtension.js [hunk 6]
```diff
@@ -39,6 +39,7 @@ module.exports = (sequelize, DataTypes) => {
 			},
 			mentee_visibility_policy: { type: DataTypes.STRING },
 			external_mentee_visibility_policy: { type: DataTypes.STRING },
+			organization_name: { type: DataTypes.STRING },
```
Comment (aks30) [inline src/database/models/organizationExtension.js:42]: @rakeshSgr rename it to just name please, Also update the corresponding DB diagrams for documentation as well.

File: src/database/migrations/20241007083318-add organisation column to OrgExt.js (hunk 4 above)
Comment (aks30) [inline src/database/migrations/20241007083318-add organisation column to OrgExt.js:11]: @rakeshSgr name can be an index if its fetched very often

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 14 Oct

File: src/services/sessions.js [hunk 7]
```diff
@@ -1225,9 +1236,14 @@ module.exports = class SessionsHelper {
...
-				userId = userDetails.id
-				email = userDetails.email
+				const userDetails = await mentorExtensionQueries.getMentorExtension(
+					userTokenData.id,
+					['user_id', 'name', 'email'],
+					true
+				)
+
+				userId = userDetails.user_id
+				email = await emailEncryption.decrypt(userDetails.email)
```
Comment (aks30) [inline src/services/sessions.js:1246]: @rakeshSgr we can put this in a common place rather than doing it everywhere maybe in pushEmailToKafka function?

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed partially on 13 Nov

File: src/utils/emailEncryption.js [hunk 8]
```diff
@@ -24,6 +24,16 @@ const decrypt = async (encryptedEmail) => {
 		throw err
 	}
 }
-const emailEncryption = { encrypt, decrypt }
+
+async function decryptAndValidate(data) {
```
[the remark this answers is not in the document]
Comment (rakeshSgr) [reply src/utils/emailEncryption.js:28]: we are trying to decrypt the email if success it returns decrypted value else it returns false

File: src/database/queries/userExtension.js [hunk 9]
```diff
@@ -462,4 +461,45 @@ module.exports = class MenteeExtensionQueries {
...
+			return results
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async getUsersByEmailIds(emailIds) {
+		try {
+			const userFilterClause =
+				emailIds.length === 0 ? '' : `email IN (${emailIds.map((id) => `'${id}'`).join(',')})`
```
[the remark this answers is not in the document]
Comment (rakeshSgr) [reply src/database/queries/userExtension.js:488]: email is not indexed. yes its encrypted. 

should we add index for email ?

File: src/requests/user.js [hunk 10]
```diff
@@ -13,25 +13,44 @@ const request = require('request')
...
  * @param {string} organisationIdentifier - The code/id of the organization.
  * @returns {Promise} A promise that resolves with the organization details or rejects with an error.
  */
 
-const fetchOrgDetails = async function ({ organizationCode, organizationId }) {
+const fetchOrgDetails = async function ({ organizationCode, organizationId, db = true }) {
 	try {
-		let orgReadUrl
+		if (organizationId && db == true) {
+			const organizationDetails = await organisationExtensionQueries.findOne({ organization_id: organizationId })
```
[the remark this answers is not in the document]
Comment (rakeshSgr) [reply src/requests/user.js:34]: currently we are not storing org code in OrganizationExtension

File: src/requests/user.js [hunk 11]
```diff
@@ -55,39 +74,59 @@ const validRoles = new Set([
...
-const fetchUserDetails = async ({ token, userId }) => {
+const fetchUserDetails = async ({ token, userId, db = true }) => {
 	try {
-		let profileUrl = `${userBaseUrl}${endpoints.USER_PROFILE_DETAILS}`
+		if (db) {
+			const userDetails = await menteeQueries.findOneFromView(userId)
 
-		if (userId) profileUrl += `/${userId}`
+			if (userDetails.image) {
+				userDetails.image = await getDownloadableUrl(userDetails.image).result
```
[the remark this answers is not in the document]
Comment (rakeshSgr) [reply src/requests/user.js:83]: added check in the getDownloadableUrl function

File: src/requests/user.js [hunk 12]
```diff
@@ -103,38 +142,90 @@ const fetchUserDetails = async ({ token, userId }) => {
...
+
+				// Map organization details for quick access
+				organizationDetails.forEach((org) => {
+					orgDetails[org.organization_id] = org
+				})
+
+				// Enrich user details with roles and organization info
+				userDetails.forEach(async (user) => {
+					if (user.image) {
+						user.image = await getDownloadableUrl(user.image).result
```
[the remark this answers is not in the document]
Comment (rakeshSgr) [reply src/requests/user.js:232]: currently we can get only one path at time from user service API

File: src/requests/user.js [hunk 13]
```diff
@@ -103,38 +142,90 @@ const fetchUserDetails = async ({ token, userId }) => {
  * @returns
  */
 
-const getListOfUserDetails = function (userIds, excludeDeletedRecords = false) {
+const getListOfUserDetails = function (userIds, db = false, excludeDeletedRecords = false) {
```
[the remark this answers is not in the document]
Comment (rakeshSgr) [reply src/requests/user.js:196]: this function has used in migrations so I kept db = false  as default.

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 20 Nov

File: src/services/mentors.js [hunk 14]
```diff
@@ -259,7 +259,7 @@ module.exports = class MentorsHelper {
 			if (session.length > 0) {
 				const userIds = _.uniqBy(session, 'mentor_id').map((item) => item.mentor_id)
 
-				let mentorDetails = await userRequests.getListOfUserDetails(userIds)
+				let mentorDetails = await userRequests.getListOfUserDetails(userIds, true)
```
[the remark this answers is not in the document]
Comment (rakeshSgr) [reply src/services/mentors.js:262]: we need update the migrations their we have used this getListOfUserDetails function.

**Pull request merged**